src/Network/WebSockets/TSSocketManager.m

```diff
@@ -259,7 +305,13 @@ - (NSString *)webSocketAuthenticationString {
 }
 
 - (void)scheduleRetry {
-    if (![self.reconnectTimer isValid]) {
+    OWSAssert([NSThread isMainThread]);
+   
+    if ([UIApplication sharedApplication].applicationState != UIApplicationStateActive) {
```

Comment:
Will this prevent us from reconnecting while running in the background due to receiving a remote notification?